tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/EmailRecoveryActionSetMethodContract.scala

```diff
@@ -539,208 +540,150 @@ trait EmailRecoveryActionSetMethodContract {
+      val deletionDateWithinUserLimit = ZonedDateTime.now().minusDays(USER_TIME_LIMIT_FOR_EMAIL_RESTORATION_IN_DAYS - 1)
+      val deletedBeforeMatching = deletionDateWithinUserLimit.plusHours(1).format(TIME_FORMATTER)
+      assertThat(restoreAndListAllMessageResult(server, BOB, deletionDateWithinUserLimit, deletedBefore = Some(deletedBeforeMatching)))
+        .hasSize(1)
+    }
 
-      val taskId: String = `given`
-        .body(
-          s"""{
-             |	"using": [
-             |		"urn:ietf:params:jmap:core",
-             |		"com:linagora:params:jmap:messages:vault"
-             |	],
-             |	"methodCalls": [
-             |		[
-             |			"EmailRecoveryAction/set",
-             |			{
-             |				"create": {
-             |					"clientId1": {
-             |						"deletedAfter": "$deletedAfter"
-             |					}
-             |				}
-             |			},
-             |			"c1"
-             |		]
-             |	]
-             |}""".stripMargin)
-      .when
-        .post
-      .`then`
-        .extract()
-        .jsonPath()
-        .get("methodResponses[0][1].created.clientId1.id").toString
+    @Test
+    def restoreShouldNotAppendMessageToMailboxWhenDeletionDateIsWithinUserLimitAndNotMatchingDeletionDateBeforeOrEquals(server: GuiceJamesServer): Unit = {
+      val deletionDateWithinUserLimit = ZonedDateTime.now().minusDays(USER_TIME_LIMIT_FOR_EMAIL_RESTORATION_IN_DAYS - 1)
+      val deletedBeforeNotMatching = deletionDateWithinUserLimit.minusHours(1).format(TIME_FORMATTER)
+      assertThat(restoreAndListAllMessageResult(server, BOB, deletionDateWithinUserLimit, deletedBefore = Some(deletedBeforeNotMatching)))
+        .hasSize(0)
+    }
 
-      awaitRestoreTaskCompleted(taskId)
+    @Test
+    def restoreShouldAppendMessageToMailboxWhenDeletionDateIsWithinUserLimitAndMatchingDeletionDateAfterOrEquals(server: GuiceJamesServer): Unit = {
+      val deletionDateWithinUserLimit = ZonedDateTime.now().minusDays(USER_TIME_LIMIT_FOR_EMAIL_RESTORATION_IN_DAYS - 1)
+      val deletedAfterMatching = deletionDateWithinUserLimit.minusHours(1).format(TIME_FORMATTER)
+      assertThat(restoreAndListAllMessageResult(server, BOB, deletionDateWithinUserLimit, deletedAfter = Some(deletedAfterMatching)))
+        .hasSize(1)
+    }
 
-      assertThat(listAllMessageResult(server, BOB)).hasSize(0)
+    @Test
+    def restoreShouldNotAppendMessageToMailboxWhenDeletionDateIsWithinUserLimitAndNotMatchingDeletionDateAfterOrEquals(server: GuiceJamesServer): Unit = {
+      val deletionDateWithinUserLimit = ZonedDateTime.now().minusDays(USER_TIME_LIMIT_FOR_EMAIL_RESTORATION_IN_DAYS - 1)
+      val deletedAfterNotMatching = deletionDateWithinUserLimit.plusHours(1).format(TIME_FORMATTER)
+      assertThat(restoreAndListAllMessageResult(server, BOB, deletionDateWithinUserLimit, deletedAfter = Some(deletedAfterNotMatching)))
+        .hasSize(0)
```

Comment:
Ok for factorizing but let's preserve a classic test structure (given .. when .. then)

And obey SRP

And find good names.